python/mirascope/llm/responses/response.py

```diff
@@ -56,6 +56,51 @@ def __init__(
             finish_reason=finish_reason,
         )
 
+    @classmethod
+    def decode(cls, data: bytes) -> "Response[None]":
+        """Deserialize a Response from JSON bytes.
+
+        Note: The decoded response will not have tools or format attached, as these
+        are code constructs that cannot be serialized. Use the decoded response's
+        messages to resume a conversation with tools if needed.
+
+        Args:
+            data: JSON-encoded bytes to decode.
+
+        Returns:
+            A Response instance reconstructed from the serialized data.
+
+        Raises:
+            IncompatibleVersionError: If the serialized data uses an incompatible version.
+            InvalidSerializedDataError: If the data is malformed.
+
+        Example:
+            ```python
+            with open("response.json", "rb") as f:
+                response = llm.Response.decode(f.read())
+            ```
+        """
+        from ..serialization import decode
+
+        return decode(data)
+
+    @classmethod
+    def decode_str(cls, data: str) -> "Response[None]":
```

Comment:
Is there a reason we'd want to have `decode` and `decode_str` be separate methods, rather than accepting `data: str | bytes` and handling both? Also, I think it should be possible to pass a `SerializedResponse` type (instance of `Jsonable`) explicitly rather than the str representation.